src/scrape.rs

```diff
@@ -0,0 +1,76 @@
+use anyhow::Result;
+use geo_types::{Point, Polygon};
+use geojson::GeoJson;
+
+/// Extract all points from a GeoJSON file.
+///
+/// TODO: Note that the returned points are not deduplicated.
+pub fn scrape_points(path: &str) -> Result<Vec<Point<f64>>> {
+    let geojson_input = fs_err::read_to_string(path)?;
+    let geojson = geojson_input.parse::<GeoJson>()?;
+    let mut points = Vec::new();
+    if let GeoJson::FeatureCollection(collection) = geojson {
+        for feature in collection.features {
+            if let Some(geom) = feature.geometry {
+                points.extend(geometry_to_points(geom.try_into().unwrap()));
+            }
+        }
+    }
+    Ok(points)
+}
+
+fn geometry_to_points(geom: geo_types::Geometry<f64>) -> Vec<Point<f64>> {
```

Comment:
Michael clued me in I can probably use https://docs.rs/geo/latest/geo/algorithm/coords_iter/trait.CoordsIter.html, I'll try this separately